File: Ruby/Factorial_iterative_no_input.rb

print "This program finds the factorial of a number.\n"

n = 0

def factorial(n)
    factorial_total = 1
    iterator = 1
    while iterator < n
        iterator += 1
        factorial_total = factorial_total * iterator
    end
    return factorial_total
end

def n_iterator(n)
    while n < 10
        n += 1
        factorial_total = factorial(n)
        print "The factorial of ", n, " is ", factorial_total, "\n"
    end
end


n_iterator(n)

File: C++/factorial_iterative_no_input.cpp


#include<iostream>
using namespace std;

int factorial(int n);
int factorial_total = 1;
int iterated_num = 1;

int main()
{
    int n = 0;
    
    while (n < 10)
    {
        n = n + 1;
        cout << "The Factorial of " << n << " = " << factorial(n) << "\n";
    }
    return 0;
}

int factorial(int n)
{

    while (iterated_num < n)
    {
        iterated_num = iterated_num + 1;
        factorial_total = factorial_total * iterated_num;
        return factorial_total;
    }

    return factorial_total;
        
}
